internal/cadence/linter.go

```diff
@@ -58,7 +58,25 @@ const (
 	ErrorCategory         = "error"
 )
 
-var analyzers = maps.Values(cdclint.Analyzers)
+// getAnalyzers returns analyzers in deterministic order (sorted by name)
+// to ensure consistent lint results across runs
+func getAnalyzers() []*analysis.Analyzer {
+	// Get keys into a slice
+	keys := make([]string, 0, len(cdclint.Analyzers))
+	for key := range cdclint.Analyzers {
+		keys = append(keys, key)
+	}
+
+	// Sort keys for deterministic order
+	slices.Sort(keys)
+
+	// Build analyzer list in sorted order
+	analyzers := make([]*analysis.Analyzer, 0, len(keys))
+	for _, key := range keys {
+		analyzers = append(analyzers, cdclint.Analyzers[key])
+	}
+	return analyzers
+}
```

Comment:
Maybe only do the sorting once at package initialization, instead of on each lint of a file:

```suggestion
var analyzers []*analysis.Analyzer

func init() {
	// Get keys into a slice
	keys := make([]string, 0, len(cdclint.Analyzers))
	for key := range cdclint.Analyzers {
		keys = append(keys, key)
	}

	// Sort keys for deterministic order
	slices.Sort(keys)

	// Build analyzer list in sorted order
	analyzers = make([]*analysis.Analyzer, 0, len(keys))
	for _, key := range keys {
		analyzers = append(analyzers, cdclint.Analyzers[key])
	}
}
```